azure/compute/src/main/java/org/jclouds/azurecompute/compute/functions/OSImageToImage.java

```diff
@@ -0,0 +1,88 @@
+ * Modifications copyright (c) Microsoft Open Technologies, Inc.
+ */
+package org.jclouds.azurecompute.compute.functions;
+
+import java.util.Map;
+
+import javax.annotation.Resource;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.azurecompute.domain.OSImage;
+import org.jclouds.compute.domain.Image;
+import org.jclouds.compute.domain.ImageBuilder;
+import org.jclouds.compute.domain.OperatingSystem;
+import org.jclouds.compute.domain.OsFamily;
+import org.jclouds.compute.domain.Image.Status;
+import org.jclouds.compute.reference.ComputeServiceConstants;
+import org.jclouds.logging.Logger;
+
+import com.google.common.base.Function;
+
+@Singleton
+public class OSImageToImage implements Function<OSImage, Image> {
+    private static String PUBLIC_CATEGORY = "Public";
+    @Resource
+    @Named(ComputeServiceConstants.COMPUTE_LOGGER)
+    protected Logger logger = Logger.NULL;
+
+    private final Map<OsFamily, Map<String, String>> osVersionMap;
+
+    @Inject
+    OSImageToImage(Map<OsFamily, Map<String, String>> osVersionMap) {
+        this.osVersionMap = osVersionMap;
+    }
+
+    protected OperatingSystem parseOs(OSImage from) {
+        OsFamily osFamily = null;
+        String osName = from.getImageFamily();  //from.getLabel();
+        String osArch = ""; //from.getImageFamily();
+        String osVersion = from.getPublishedDate() == null ? null : "" + from.getPublishedDate().getTimeInMillis();
+        String osDescription = ""; //from.getDescription() == null ? "" : from.getDescription();
+        try {
+            osFamily = OsFamily.fromValue(from.getOSType().toLowerCase());
+        } catch (IllegalArgumentException e) {
+            logger.debug("<< didn't match os(%s)", from.getName());
+        }
+        boolean is64Bit = true;
+//                (from.getName() != null && from.getName().contains("64")) ||
+//                (from.getDescription() != null && from.getDescription().contains("64")) ||
+//                (from.getLabel() != null && from.getLabel().contains("64")) || osFamily == OsFamily.WINDOWS;
+
+        return new OperatingSystem(osFamily, osName, osVersion, osArch, osDescription, is64Bit);
+    }
+
+    @Override
+    public Image apply(OSImage from) {
+        ImageBuilder builder = new ImageBuilder();
+        builder.ids(from.getName());
+        builder.name(from.getLabel());
+        builder.description(from.getDescription() == null ? "" : from.getDescription());
+        builder.operatingSystem(parseOs(from));
+        builder.uri(from.getMediaLink());
+//        builder.status(toPortableImageStatus.get(from.getState()));
+        builder.status(Status.AVAILABLE);
+        builder.status(PUBLIC_CATEGORY.equals(from.getCategory()) ? Status.AVAILABLE : Status.PENDING);
+        builder.userMetadata(ImmutableMap.<String, String> of(OSImage.LOCATION, from.getLocation()));
```

Comment:
Can this be transformed into a proper location and set the location field properly?